src/Utilities/Compiler/RuleLevel.cs

```diff
@@ -47,6 +47,12 @@ internal enum RuleLevel
         /// <b>Disabled by default rule</b>, which is a candidate for <b>deprecation</b>.
         /// </summary>
         CandidateForRemoval = 5,
+
+        /// <summary>
+        /// Correctness rule which should have <b>no false positives</b>, and is required to be fixed by users.
+        /// This rule will be <b>enabled in CI and IDE live analysis</b> by default with severity <see cref="DiagnosticSeverity.Error"/>.
+        /// </summary>
+        BuildError = 6,
```

Comment:
I think it is better to give this value 1 and increment the rest of the fields, the enum goes from highest to lowest severity and its not a public API so a breaking change is fine.